### PR TITLE
[API CONFORMANCE] [TEMPLATE PLUGIN] `CachingModelTests` 

### LINKED ISSUE #21772: [Good first issue]: [API CONFORMANCE] [TEMPLATE PLUGIN] `CachingModelTests` partually are failing on not-supported operation in interpreter
## Description:
API conformance suite is OV validation tool checks a plugin conditions from API implementation perspective.
`ov_plugin*CompileModelCacheTestBase*` fails on TEMPLATE Plugin` fails on TEMPLATE plugin according not supported operations in interpreter backend.

## How to reproduce:
1. Build the OV using `-DENABLE_TESTS=ON -DENABLE_FUNACTIONAL_TESTS=ON`
2. Build `ov_api_conformance_tests` target
3. run API conformance: `ov_api_conformance_tests --gtest_filter=*ov_plugin_numeric*CompileModelCacheTestBase* --device=TEMPLATE`
4. Get the result for unsupported properties:
```
[ RUN      ] ov_plugin_numeric/CompileModelCacheTestBase.CompareWithRefImpl/SimpleFunctionMultiply_u16_batch1_TEMPLATE

MEM_USAGE=75876KB
Access PluginCache ov core. OV Core use count: 0
Created ov core.

MEM_USAGE=80980KB
[ CONFORMANCE ] Influence coefficient: 0
/home/efode/repo/openvino/src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp:241: Failure
Failed
Can't loadNetwork without cache for SimpleFunctionMultiply with precision u16
Exception [Exception from src/inference/src/infer_request.cpp:231:
Check 'it != map.end()' failed at src/plugins/template/backend/int_executable.cpp:212:
Interpreter backend doesn't implement evaluate method for OP Multiply

]

Reset PluginCache. OV Core use count: 1
[  FAILED  ] ov_plugin_numeric/CompileModelCacheTestBase.CompareWithRefImpl/SimpleFunctionMultiply_u16_batch1_TEMPLATE, where GetParam() = ((32-byte object <D3-11 68-67 F2-55 00-00 2C-20 22-54 45-4D 50-4C 79-D0 69-67 F2-55 00-00 EE-CF 69-67 F2-55 00-00>, "SimpleFunctionMultiply"), u16, 1, "TEMPLATE", {}) (83 ms)
```

## Useful links:
* [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
* [Conformance readme](https://github.com/openvinotoolkit/openvino/tree/master/src/tests/functional/plugin/conformance/test_runner)
* [TEMPLATE plugin](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/template/README.md)
* [OV CONTRIBUTION GUIDE](https://github.com/openvinotoolkit/openvino/tree/master/?tab=readme-ov-file#how-to-contribute)

## Contacts:
* @iefode @sbalandi

### PR DESCRIPTION
**Pull request by LucaTamSapienza**
### Tickets:
Closes #21772

### CODE REVIEW

File: src/plugins/template/backend/ops/multiply.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+template <ov::element::Type_t ET>
+bool evaluate(const std::shared_ptr<ov::op::v1::Multiply>& op, ov::TensorVector& outputs, const ov::TensorVector& inputs) {
+    using T = typename ov::element_type_traits<ET>::value_type;
+    ov::reference::multiply<T>(inputs[0].data<T>(), inputs[1].data<T>(), outputs[0].data<T>(), ov::shape_size(inputs[0].get_shape()));
+    return true;
+}
+
+template <>
+bool evaluate_node<ov::op::v1::Multiply>(std::shared_ptr<ov::Node> node,
```
Comment (praasz) [inline src/plugins/template/backend/ops/multiply.cpp:17]: The `v1::Multiply` has `evalute` function and support some types [see](https://github.com/openvinotoolkit/openvino/blob/3238290df5ecb9e0eb0c4254d20bf668c3bc1408/src/core/src/op/multiply.cpp#L62).
The template implementation can just   and this implementation can be just extension of required types not all are required (remove duplicates)

The types such as `i4`, `u4`, `u1` can be removed as reference will not support them correctly.
The `boolean` type can be removed as is not legal value for this operator.

Consider remove types which duplicate list in `ov::op::v1::Multiply::has_evaluate` and which cannot be handled correctly by reference implementation.

Comment (LucaTamSapienza) [reply src/plugins/template/backend/ops/multiply.cpp:17]: @praasz 
Thank you very much for your code review! I’ll remove all the duplicates in the next hours.

File: src/plugins/template/backend/ops/multiply.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                                         ov::TensorVector& outputs,
+                                         const ov::TensorVector& inputs) {
+    auto element_type = node->get_output_element_type(0);
+
+    switch (element_type) {
+    case ov::element::boolean:
+        return evaluate<ov::element::boolean>(ov::as_type_ptr<ov::op::v1::Multiply>(node), outputs, inputs);
+    case ov::element::bf16:
```
Comment (iefode) [inline src/plugins/template/backend/ops/multiply.cpp:25]: According @praasz comment, could you please leave only types which are not supported in `evaluate` method of `Multiply`

**Comment by iefode** [comment]
@LucaTamSapienza Thanks for great work!
Please apply comments and we will merge it

**Comment by praasz** [comment]
build_jenkins

**Review by praasz — APPROVED** [review]
Fix the code style, please.

**Comment by LucaTamSapienza** [comment]
@praasz 
Unfortunately, it's not the first time that I can't close a PR due to the code style. I followed the instructions on this [page](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/coding_style.md), but since I have the latest version of Ubuntu (22.04), the clang-format-9 package is not available for it. I've searched everywhere to try to install it, but without success. If you have any suggestions, I would gladly accept them

File: src/plugins/template/backend/ops/ops_evaluates.hpp [hunk 3]
```diff
@@ -175,6 +175,10 @@ extern template bool evaluate_node<ov::op::v1::Mod>(std::shared_ptr<ov::Node> no
                                                     ov::TensorVector& outputs,
                                                     const ov::TensorVector& inputs);
 
+extern template bool evaluate_node<ov::op::v1::Multiply>(std::shared_ptr<ov::Node> node,
+                                                        ov::TensorVector& outputs,
+                                                        const ov::TensorVector& inputs);
+
```
Comment (praasz) [inline src/plugins/template/backend/ops/ops_evaluates.hpp:181]:
```suggestion
extern template bool evaluate_node<ov::op::v1::Multiply>(std::shared_ptr<ov::Node> node,
                                                         ov::TensorVector& outputs,
                                                         const ov::TensorVector& inputs);

```

File: src/plugins/template/backend/ops/multiply.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+#include "openvino/reference/multiply.hpp"
+
+#include "evaluate_node.hpp"
+
+template <ov::element::Type_t ET>
+bool evaluate(const std::shared_ptr<ov::op::v1::Multiply>& op, ov::TensorVector& outputs, const ov::TensorVector& inputs) {
+    using T = typename ov::element_type_traits<ET>::value_type;
+    ov::reference::multiply(inputs[0].data<const T>(), inputs[1].data<const T>(), outputs[0].data<T>(), ov::shape_size(inputs[0].get_shape()));
```
Comment (praasz) [inline src/plugins/template/backend/ops/multiply.cpp:12]:
```suggestion
bool evaluate(const std::shared_ptr<ov::op::v1::Multiply>& op,
              ov::TensorVector& outputs,
              const ov::TensorVector& inputs) {
    using T = typename ov::element_type_traits<ET>::value_type;
    ov::reference::multiply(inputs[0].data<const T>(),
                            inputs[1].data<const T>(),
                            outputs[0].data<T>(),
                            ov::shape_size(inputs[0].get_shape()));
```

**Comment by praasz** [comment]
> @praasz Unfortunately, it's not the first time that I can't close a PR due to the code style. I followed the instructions on this [page](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/coding_style.md), but since I have the latest version of Ubuntu (22.04), the clang-format-9 package is not available for it. I've searched everywhere to try to install it, but without success. If you have any suggestions, I would gladly accept them

Ok, I can propose formatted suggestion. I'll hope that fix CI.

**Comment by LucaTamSapienza** [comment]
@praasz  
Thank you very much for the help. I think I'll do a backup and return to Ubuntu 20.04, which is definitely more stable.

**Comment by praasz** [comment]
build_jenkins

**Review by iefode — APPROVED** [review]

**Pull request merged**